GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1563,7 +1563,8 @@ private void OnShowSettingsClick(object sender, EventArgs e)
             RevisionGrid.ReloadTranslation();
             fileTree.ReloadHotkeys();
             revisionDiff.ReloadHotkeys();
-            revisionDiff.ReloadCustomDifftools();
+            revisionDiff.LoadCustomDifftools();
+            GitModule.GitCommandCache.Clear();
```

Comment:
Shall the cache be cleared after loading? 🤔

Reply:
Not really, the entries in the cache should be unique from repo to repo (assuming shas are unique). Tools could be per repo too, that will require a reload, it is a reasonable limitation in my opinion.

Reply:
This is executed after closing FormSettings. Why not clear the cache before loading the difftools?
Why have you added `.Clear()` at all? Is it fixing another issue? I agree to "should be unique from repo to repo (assuming shas are unique)". How is it related to difftools?

---

Can the initial loading of the difftools be postponed until the RevisionGrid has been populated?
My AV virus becomes very active and slows down the more important git commands.

Reply:
Without a cache, it can take over 10s for me (with Symantec virus) before the tools are available.
This is OK for FormBrowse, but not for FormCommit or ResolveConflicts.

If a user add or remove a tool it is not updating the list unless the cache is cleared.
We can of course require that the user close/open GE to refresh the list, it is not unreasonable.
But with a GE centric perspective, difftools are added and removed in Settings, I believe the user expects that a newly added tool is available in th lst if added in settings.
This can of course be described in the settings tooltip (that is to be added).

Reply:
I am not arguing neither against the cache nor against clearing it after settings have changed.

But I have not got yet why not rather use the following sequence: 
```
            GitModule.GitCommandCache.Clear();
            revisionDiff.LoadCustomDifftools();
```

Reply:
Of course, this order will be changed